### PR TITLE
File handling and sample files

### PR DESCRIPTION
**Pull request by ericrommel**

### CODE REVIEW

File: weeks/week03/file-handling.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+with open("my_test_data.txt", "a") as file:
+    file.write("Appending this line.")
+````
+
+
+## 1.4 File Handling Exceptions
+
+Handling exceptions is crucial for error-prone operations like file handling.
+
+When using the context manager `with` in file handling, the context manager itself handles the automatic opening and closing of the file, even if an exception occurs during the operation. However, it does not handle exceptions related to file operations (like missing files or permission errors). Therefore, using a `try-except` block is necessary if you want to catch and handle exceptions such as `FileNotFoundError`, `PermissionError`, or other file-related errors.
```
Comment (SSN-SPB) [inline weeks/week03/file-handling.md:124]: imho #1.4 should be combined with ## 1.9 Handling File Exceptions Gracefully in one place (e.g. for mention all FileHandling related exception)

**Review by SSN-SPB — CHANGES_REQUESTED** [review]
Generally it looks good. The only one comment. The one substantial part should be mentioned at least: directory handling.

**Comment by ericrommel** [comment]
> Generally it looks good. The only one comment. The one substantial part should be mentioned at least: directory handling.

I would skip to talk about it for two reasons:
- We are covering the main topic: file handling
- If they need to handle directories, they would be able to do it by themselves as we covered file handling.

File: weeks/week03/file-handling.md (hunk 1 above)
Comment (ericrommel) [reply weeks/week03/file-handling.md:127]: Done!

**Review by SSN-SPB — APPROVED** [review]
LGTM

**Pull request merged**